src/uct/api/uct_def.h

```diff
@@ -111,6 +117,13 @@ typedef int                      uct_worker_cb_id_t;
     |<---- stride ----->|<---- stride ----->|
    @endverbatim
  *
+ * In order to describe an interleaved memory layout, each structure contains
```

Comment:
There are problems with this proposal:
1. memh_ABAB exceeds the bounds of its underlying memh_AB, which makes bounds check complicated & costly (recursive down) for what is supposed to be practically data-path. 
2. This hurts the performance of UMR since you need more calls to ep_mem_reg before you can start sending
3. It complicates the user that now needs more memh-s for the same pattern
4. The creation of the UMR also takes more time because you need to combine multiple datatype structs to post the WR (the WR for memh_ABAB will not point to memh_AB - it needs to point to memh_A and memh_B directly, otherwise we have overhead on the HCA filling each send request)

The only benefit I see to this is removing the "extra" interleaving ratio field from the API... Is it worth it?